Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/DriverOnboarding/HyperVergeWebhook.hs

```diff
@@ -0,0 +1,60 @@
+module Domain.Action.UI.DriverOnboarding.HyperVergeWebhook where
+
+import Data.Aeson
+import qualified Data.Text as T
+import qualified Domain.Types.DocumentVerificationConfig as DVC
+import Environment
+import Kernel.Prelude
+import qualified Kernel.Types.Documents as Documents
+import Kernel.Utils.Common hiding (Error)
+import Servant hiding (throwError)
+import qualified Storage.Queries.AadhaarCard as QAadhaarCard
+import qualified Storage.Queries.DriverPanCard as QDPC
+import qualified Storage.Queries.Image as QImage
+import Tools.Error
+
+type HyperVergeResultWebhookAPI =
+  "hyperverge"
+    :> "result"
+    :> ReqBody '[JSON] Value
+    :> Post '[JSON] AckResponse
+
+data HyperVergeResultWebhookPayoad = HyperVergeResultWebhookPayoad
+  { transactionId :: Text,
+    applicationStatus :: Text,
+    eventId :: Maybe Text,
+    eventVersion :: Maybe Text,
+    eventTime :: Maybe Text,
+    eventType :: Maybe Text,
+    reviewerEmail :: Maybe Text
+  }
+  deriving (Generic, ToJSON, FromJSON, Show, Eq, ToSchema)
+
+hyperVergeResultWebhookHandler :: Value -> Flow AckResponse
+hyperVergeResultWebhookHandler payload = do
+  logDebug $ "HyperVerge Result Webhook Payload: " <> show payload
+  parsedPayload <- case (fromJSON payload :: Result HyperVergeResultWebhookPayoad) of
+    Error err -> throwError $ InvalidWebhookPayload "HyperVerge" (T.pack err)
+    Success pyload -> return pyload
+  when (isNothing parsedPayload.reviewerEmail) $ logError "Warning !!!!! Missing Field reviewerEmail in HyperVerge webhook Payload!!!!. Continuing the flow as other necessary fields are present."
+  vstatus <- convertHVStatusToPanValidationStatus parsedPayload.applicationStatus
+  QImage.updateVerificationStatus (Just vstatus) parsedPayload.reviewerEmail (Just parsedPayload.transactionId)
+  imageEntity <- QImage.findByWrokflowTransactionId (Just parsedPayload.transactionId) >>= fromMaybeM (ImageNotFoundForWorkflowId parsedPayload.transactionId) . listToMaybe
```

Comment:
findAll or findOne?

Reply:
as we can have multiple entries with same workflowTransactionId we should use findAll as findOne will break.